docs/resources/server/enchantments.md

```diff
@@ -0,0 +1,85 @@
+```java
+
+for(Enchantment enchant : stack.get(DataComponents.ENCHANTMENTS).keySet()){
+    DataComponentMap enchantment_components = enchant.value().effects();
+    if(enchantment_components.has(MY_ENCHANTMENT_COMPONENT_TYPE_HOLDER)){
+            // Do something!
+    }
+}
+```
+
+## Specialized Enchantment Component Types
+Vanilla uses a few specific classes when building its enchantment component types to achieve the customizability present in the vanilla enchantment components.
+
+### `ConditionalEffect`
+Wrapping the type in `List<ConditionalEffect<?>>` permits the use of [predicates] when defining enchantments. This allows some effects to activate situationally. Most vanilla enchantments use this.
+Registering such an effect can be done as follows:
+```java
+public static final DeferredHolder<DataComponentType<?>, DataComponentType<List<ConditionalEffect<Unit>>>> EXAMPLE_CONDITIONAL_EFFECT =
+    ENCHANTMENT_COMPONENT_TYPES.register("example_conditional",
+        () -> DataComponentType.<List<ConditionalEffect<Unit>>>builder()
+            .persistent(ConditionalEffect.codec(Unit.CODEC, LootContextParamSets.EMPTY).listOf())
+            .build());
+```
+The parameters to `ConditionalEffect.codec` are the codec for the template type of the `ConditionalEffect<T>`, followed by some `LootContextParamSets` item. There are many of these to choose from, so choose carefully.
+
+### Value Effects
+[Value Effects] are used for enchantments that change some numerical value to a different degree depending on the enchantment's level, and are implemented by the class `EnchantmentValueEffect`. Enchantments like Knockback, Looting, an Sharpness use this kind of component.
+
+A vanilla-conforming way to adjust values based on custom Value Effect components is to invoke one of the overloads of `EnchantmentHelper#runIterationOnItem` and pass in a lambda that alters a `MutableFloat` or similar mutable object based on the results:
+```java
+MutableFloat mutable_value = new MutableFloat(unmodified_value);
+EnchantmentHelper.runIterationOnItem(item_stack, (enchantment, enchant_level) -> Enchantment.applyEffects(
+    enchantment.value().getEffects(MY_ENCHANTMENT_COMPONENT_TYPE_HOLDER.value()),
+    // Provide a loot context,
+    (value_effect) -> mutable_value.setValue(value_effect.process(enchant_level, server.random, mutable_value.getValue()))
+));
+```
+
+Custom numerical operations for use in Value Enchantment blocks can be added by registering a subclass of `EnchantmentValueEffect` through `BuiltInRegistries.ENCHANTMENT_VALUE_EFFECT_TYPE`.
+
+### Entity Effects
+[Entity Effects] are used for enchantments that cause special effects to either the wielder or (if applicable) the target of an attack when the enchanted item is used to attack another entity. Enchantments like Flame, Fire Aspect, and Channeling use this kind of component.
+
+Custom `EnchantmentEntityEffect` extensions can be registered through `BuiltInRegistries.ENCHANTMENT_ENTITY_EFFECT_TYPE`.
```

Comment:
This is worded in a misleading way. The enchantment entity effect is just a data object that can be a part of an enchantment effect component. For example, there are 'entity effects' for after the entity finishes an attack or hits a block.

It's a separate interface to make it easier to implement how that effect is applied to the entity, rather than writing it out in its place of usage, since it could be used in multiple places.